aspnetcore/fundamentals/localization/sample/Program.cs

```diff
@@ -1,24 +1,146 @@
-﻿using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Linq;
-using System.Threading.Tasks;
+﻿using System.Globalization;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Localization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Localization.StarterWeb.Models;
+using Localization.StarterWeb.Services;
 
 namespace Localization.StarterWeb
 {
     public class Program
     {
+        public static IConfiguration Configuration { get; set; }
+        public static IHostingEnvironment Environment { get; set; }
+
         public static void Main(string[] args)
         {
-            var host = new WebHostBuilder()
-                .UseKestrel()
-                .UseContentRoot(Directory.GetCurrentDirectory())
-                .UseIISIntegration()
-                .UseStartup<Startup>()
-                .Build();
-
-            host.Run();
+            BuildWebHost(args).Run();
         }
+
+        public static IWebHost BuildWebHost(string[] args) =>
+            WebHost.CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration((hostingContext, config) =>
+                {
+                    Configuration = config.Build();
+                    Environment = hostingContext.HostingEnvironment;
+                })
+                .ConfigureServices(services =>
```

Comment:
This is not the style we usually use, see [here](https://github.com/aspnet/Mvc/blob/dev/samples/MvcSandbox/Startup.cs) for an example of our recommended structure.

Reply:
*That's very 1.x of you!* LOL :grinning: AFAIK, we're trying (or "I'm trying" :smile:) to use the `CreateDefaultBuilder` convenience method (along with `ConfigureAppConfiguration` where needed) to centralize and simplify the ceremony of the startup/config pattern (that you linked, for example). It's nice and neat.

This is really up to the boss, of course. We can sit on this until @Rick-Anderson can TAL (or perhaps @scottaddie knows from the internal meetings what they decided).

I 😍 this startup pattern, so this is what I would do in my own stuff, and this is what I hope MS will adopt everywhere. However, the *NutJob*:tm: startup pattern used by silly green dinosaurs isn't always so well received. I'll change it back if those are the orders.